ReactiveCocoa/Swift/DynamicProperty.swift

```diff
@@ -1,29 +1,38 @@
 import Foundation
 import enum Result.NoError
 
+/// Models types that can be represented in Objective-C (i.e., reference
+/// types, including generic types when boxed via `AnyObject`).
+private protocol ObjectiveCRepresenting {
+	associatedtype Value
+	func extractValue(fromRepresentation representation: AnyObject) -> Value
+	func represent(value value: Value) -> AnyObject
+}
+
 /// Wraps a `dynamic` property, or one defined in Objective-C, using Key-Value
 /// Coding and Key-Value Observing.
 ///
 /// Use this class only as a last resort! `MutableProperty` is generally better
 /// unless KVC/KVO is required by the API you're using (for example,
 /// `NSOperation`).
-@objc public final class DynamicProperty: RACDynamicPropertySuperclass, MutablePropertyType {
-	public typealias Value = AnyObject?
+public final class DynamicProperty<Object>: MutablePropertyType {
```

Comment:
I've just rebased and pushed two rewritten commits that no longer use `Object`. The main reason I used something other than `Value` was this code:

```swift
return object?.valueForKeyPath(keyPath).map { $0 as! Value }
```

In that situation, `Value` reads as ambiguous: is this the optional or non-optional `Value`? In practice it's the optional `Value?`, which shadows the generic parameter `Value`.

In the new version I used `SwiftValue` so that I could refer to the type parameter explicitly (as opposed to the typealias), and then ultimately was able to remove that in favour of the implicitly defined `Value` typealias. Let me know what you think.